### ISSUE TITLE
Proposal: Move subprojects to their own repositores

### ISSUE THREAD

**Issue by ademar** [SuaveIO/suave#581]
and build independently.

**Comment by haf** [comment]
I think that's a bad idea for the following reasons:

 - As I've experienced with my (non-/open-)source work, there's a significant overhead to .net development in multiple projects
 - Subprojects will inevitably go out of sync
 - There's no cascading build tool available for .Net like there is for Haskell
 - It increases confusion about where to send a pull request
 - Things that are annoying to get right, like .Net core builds would be N times more annoying to get right in subprojects

The only upside I see is that releases like the one I just did won't be needed, because Suave.Testing is revved on its own. This brings a downside:

 - The compatibility matrix between the Suave subprojects starts to diverge and people will have to "guess" which versions of the different Suave projects belong together.

I've had a similar split between Logary and the commercial targets and there, the commercial targets are constantly out of date.

Also, due to the slowness of the F# compiler, having multiple projects decreases programmer efficiency; in between building, releasing and updating consecutive/dependent sub projects. Instead of a single command (see the lack of a cascading build tool argument), there are now multiple commands.

Finally, NuGet.org is eventually consistent, meaning that subprojects cannot directly reference new releases, further eroding programmer efficiency.
